src/main/java/com/xiaomi/infra/pegasus/rpc/interceptor/SecurityReplicaSessionInterceptor.java

```diff
@@ -32,22 +37,32 @@
 
   private String serviceName;
   private String serviceFqdn;
+
+  // JAAS internal class, Ref:
+  // https://docs.oracle.com/javase/7/docs/technotes/guides/security/jaas/JAASRefGuide.html
+  //
+  // To authorize access to resources, applications first need to authenticate the source of the
+  // request. The JAAS framework defines the term subject to represent the source of a request. A
+  // subject may be any entity, such as a person or a service
   private Subject subject;
+  // The LoginContext class provides the basic methods used to authenticate subjects, and provides a
+  // way to develop an application independent of the underlying authentication technology
   private LoginContext loginContext;
 
-  public SecurityReplicaSessionInterceptor(String serviceName, String serviceFqdn)
-      throws IllegalArgumentException {
-    this.serviceName = serviceName;
-    this.serviceFqdn = serviceFqdn;
+  public SecurityReplicaSessionInterceptor(ClientOptions options) throws IllegalArgumentException {
+    this.serviceName = options.getServiceName();
+    this.serviceFqdn = options.getServiceFQDN();
 
     try {
-      loginContext = new LoginContext("client", new TextCallbackHandler());
+      // actual authentication: authenticate the Subject
+      subject = new Subject();
+      loginContext =
+          new LoginContext(
+              "pegasus-client",
+              subject,
+              new TextCallbackHandler(),
```

Comment:
I've no found any refs why use `TextCallbackHandler`. Could you explain it in the comment?